docs/manual/src/internals/object_references.md

```diff
@@ -32,8 +32,8 @@ this is so.
 ## Lifetimes
 
 In order to allow for instances to be used as flexibly as possible from foreign-language code,
```

Comment:
This duplicates some of the docs above - I'd suggest maybe merging the above with this - ie, this can be the "advanced" section I referred to.